bin/docker-image-tool.sh

```diff
@@ -71,18 +71,29 @@ function build {
     --build-arg
     base_img=$(image_ref spark)
   )
-  local BASEDOCKERFILE=${BASEDOCKERFILE:-"$IMG_PATH/spark/Dockerfile"}
-  local PYDOCKERFILE=${PYDOCKERFILE:-"$IMG_PATH/spark/bindings/python/Dockerfile"}
-  local RDOCKERFILE=${RDOCKERFILE:-"$IMG_PATH/spark/bindings/R/Dockerfile"}
+  local BASEDOCKERFILE=${BASEDOCKERFILE:-"$IMG_PATH/main/dockerfiles/spark/Dockerfile"}
+  local PYDOCKERFILE=${PYDOCKERFILE:-"$IMG_PATH/main/dockerfiles/spark/bindings/python/Dockerfile"}
+  local KDOCKERFILE=${KDOCKERFILE:-"$IMG_PATH/test/dockerfiles/spark/kerberos/Dockerfile"}
+  local RDOCKERFILE=${RDOCKERFILE:-"$IMG_PATH/main/dockerfiles/spark/bindings/R/Dockerfile"}
 
+  # Spark Base
   docker build $NOCACHEARG "${BUILD_ARGS[@]}" \
     -t $(image_ref spark) \
     -f "$BASEDOCKERFILE" .
 
+  # PySpark
   docker build $NOCACHEARG "${BINDING_BUILD_ARGS[@]}" \
     -t $(image_ref spark-py) \
     -f "$PYDOCKERFILE" .
 
+  # The following are optional docker builds for Kerberos Testing
+  docker pull ifilonenko/hadoop-base:latest
```

Comment:
Ur, this looks a little bit too personal. is this okay for Apache Spark?

Reply:
What about using `docker build` here instead?

Reply:
I comment on this earlier... https://github.com/apache/spark/pull/22608#issuecomment-427617701

Reply:
Thanks. Got it, @felixcheung .

Reply:
In fact, `FROM` has the same problem here. What I mean was including the whole docker file from `ifilonenko/hadoop-base`. Although this is a test script, I was surprised that Apache Spark depends on a personal repo officially.

Reply:
@ifilonenko why do you need to pull that image here?

Reply:
I had also discussed concerns re: pulling a "personal" image w/ @ifilonenko - I'd prefer some other solution here. Docker build seems plausible, but it would be best if it can be done once per test run for efficiency reasons

Reply:
The strategy of using this “personal” image is purely because the HDFS nodes: (kdc, nn, dn) themselves require hadoop-2.7.3.tgz. Unless that can be packaged in the distribution, in which case it would be trivial to build, pulling is easier. I am open for opinions on building strategies. The dockerfile in question can be found here: https://github.com/ifilonenko/hadoop-kerberos-helm/blob/master/Dockerfile